src/bit-fields/src/lib.rs

```diff
@@ -0,0 +1,1557 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A macro to generate structures which support bit flags and sub-bytes ranges.
```

Comment:
Isn t it possible to break this 1500 file into logical bits?

Maybe create a different file for all features we implement and move the logic there?

Reply:
I'm not sure if you are referring to `bit-fields/src/lib.rs` file as this is 336 lines.

Reply:
I have moved some of the error types to a different file. This unfortunately only has a small impact, I cannot see a good way to significantly reduce the line length here.